src/api/procedures/__tests__/utils.ts

```diff
@@ -0,0 +1,102 @@
+import { Call } from '@polkadot/types/interfaces/runtime';
+
+import { ProposalStage } from '~/api/entities/Proposal/types';
+import { Context } from '~/context';
+import { dsMockUtils, entityMockUtils } from '~/testUtils/mocks';
+import { Mocked } from '~/testUtils/types';
+
+import { assertProposalUnlocked } from '../utils';
+
+jest.mock(
+  '~/api/entities/Proposal',
+  require('~/testUtils/mocks/entities').mockProposalModule('~/api/entities/Proposal')
+);
+
+describe('assertProposalUnlocked', () => {
+  const pipId = 10;
+  const mockAddress = 'someAddress';
+
+  let mockContext: Mocked<Context>;
+
+  beforeAll(() => {
+    dsMockUtils.initMocks({
+      contextOptions: {
+        currentPairAddress: mockAddress,
+      },
+    });
+    entityMockUtils.initMocks();
+  });
+
+  beforeEach(() => {
+    mockContext = dsMockUtils.getContextInstance();
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    entityMockUtils.cleanup();
+    dsMockUtils.cleanup();
+  });
+
+  test('should throw an error if the proposal is not in pending state', async () => {
+    entityMockUtils.configureMocks({
+      proposalOptions: {
+        getDetails: dsMockUtils.createMockPip({
+          id: dsMockUtils.createMockU32(),
+          proposal: ('proposal' as unknown) as Call,
+          state: dsMockUtils.createMockProposalState('Killed'),
+        }),
+      },
+    });
+
+    let error;
+    try {
+      await assertProposalUnlocked(pipId, mockContext);
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error.message).toBe('The proposal must be in pending state');
+  });
+
+  test('should throw an error if the proposal is not in the cool off period', async () => {
+    entityMockUtils.configureMocks({
+      proposalOptions: {
```

Comment:
You should specifically set a stage here for it to be more clear why the test passes